app/kuma-dp/pkg/dataplane/envoy/dpserver.go

```diff
@@ -0,0 +1,75 @@
+package envoy
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/kumahq/kuma/pkg/core"
+	"github.com/kumahq/kuma/pkg/core/runtime/component"
+)
+
+var _ component.GracefulComponent = &DPServerComponent{}
+
+var dpServerLog = core.Log.WithName("kuma-dp").WithName("run").WithName("dpServer")
+
+type DPServerComponent struct {
+	mux         *http.ServeMux
+	port        int
+	notifyDrain chan<- struct{}
+	done        chan struct{}
+}
+
+func NewDPServer(port int, notifyDrain chan<- struct{}) component.GracefulComponent {
```

Comment:
I understand this `DPServerComponent` is designed to be a generic server container that can hold arbitrary logic that is fired by this channel. If this is the idea, should we name the `notifyDrain` to something else, like "triggerer". In addition, the the http path and method should also be generic to make this a generic container. 
Otherwise, if we don't make it generic, should we put the calling to the `envoyComponent.DrainForever` into this class?